internal/controller/proxmoxcluster_controller.go

```diff
@@ -168,6 +168,22 @@ func (r *ProxmoxClusterReconciler) reconcileNormal(ctx context.Context, clusterS
 	// If the ProxmoxCluster doesn't have our finalizer, add it.
 	ctrlutil.AddFinalizer(clusterScope.ProxmoxCluster, infrav1alpha1.ClusterFinalizer)
 
+	if clusterScope.ProxmoxCluster.Spec.ExternalManagedControlPlane {
+		if clusterScope.ProxmoxCluster.Spec.ControlPlaneEndpoint.Host == "" {
```

Comment:
I think we should add a check,  `if clusterScope.ProxmoxCluster.Spec.ControlPlaneEndpoint != nil`

Reply:
good catch - it actually panicked the test